src/views/IncidentMap.js

```diff
@@ -110,6 +113,16 @@ const IncidentMap = (props) => {
         }
     }, [selectedState])
 
+    // Update legends when they're first created
+    useEffect(() => {
+        if (mapLegend) {
```

Comment:
no need to check if mapLegen is valid or not. line 101 and 102 did not do any check. let's make it consistent.